### PR TITLE
add completablefuture annotations

### PR DESCRIPTION
**Pull request by J-N-K**
Signed-off-by: Jan N. Klug <jan.n.klug@rub.de>

### CODE REVIEW

**Comment by jason-faust** [comment]
I'll take a look at these closer to the weekend when I have a chance to get a working environment for this stuff again.

**Comment by jason-faust** [comment]
Which Java version is this for?  11?

**Comment by J-N-K** [comment]
Yes.

File: libraries/java/java/util/concurrent/CompletableFuture.eea [hunk 1]
```diff
@@ -26,18 +26,51 @@ applyToEitherAsync
...
+failedFuture
+ <U:Ljava/lang/Object;>(Ljava/lang/Throwable;)Ljava/util/concurrent/CompletableFuture<TU;>;
+ <U:Ljava/lang/Object;>(L1java/lang/Throwable;)L1java/util/concurrent/CompletableFuture<TU;>;
+failedStage
+ <U:Ljava/lang/Object;>(Ljava/lang/Throwable;)Ljava/util/concurrent/CompletionStage<TU;>;
+ <U:Ljava/lang/Object;>(L1java/lang/Throwable;)L1java/util/concurrent/CompletionStage<TU;>;
 get
  (JLjava/util/concurrent/TimeUnit;)TT;
- (JL1java/util/concurrent/TimeUnit;)TT;
+ (JL1java/util/concurrent/TimeUnit;)T0T;
```
Comment (jason-faust) [inline libraries/java/java/util/concurrent/CompletableFuture.eea:70]: I'd revert this to what it was previously and let the declaration of T handle the nullness of the return.

File: libraries/java/java/util/concurrent/CompletableFuture.eea [hunk 2]
```diff
@@ -26,18 +26,51 @@ applyToEitherAsync
...
+getNow
+ (TT;)TT;
+ (T0T;)T0T;
```
Comment (jason-faust) [inline libraries/java/java/util/concurrent/CompletableFuture.eea:73]: (TT;)TT;
Allow the declaration of T handle the nullness

File: libraries/java/java/util/concurrent/CompletableFuture.eea [hunk 3]
```diff
@@ -26,18 +26,51 @@ applyToEitherAsync
 complete
  (TT;)Z
  (T0T;)Z
```
Comment (jason-faust) [inline libraries/java/java/util/concurrent/CompletableFuture.eea:28]: (TT;)Z

File: libraries/java/java/util/concurrent/CompletableFuture.eea [hunk 4]
```diff
@@ -26,18 +26,51 @@ applyToEitherAsync
...
  (L1java/lang/Throwable;)Z
+completeOnTimeout
+ (TT;JLjava/util/concurrent/TimeUnit;)Ljava/util/concurrent/CompletableFuture<TT;>;
+ (TT;JL1java/util/concurrent/TimeUnit;)L1java/util/concurrent/CompletableFuture<TT;>;
 completedFuture
  <U:Ljava/lang/Object;>(TU;)Ljava/util/concurrent/CompletableFuture<TU;>;
  <U:Ljava/lang/Object;>(T0U;)L1java/util/concurrent/CompletableFuture<TU;>;
+completedStage
+ <U:Ljava/lang/Object;>(TU;)Ljava/util/concurrent/CompletionStage<TU;>;
+ <U:Ljava/lang/Object;>(T0U;)L1java/util/concurrent/CompletionStage<TU;>;
```
Comment (jason-faust) [inline libraries/java/java/util/concurrent/CompletableFuture.eea:46]: <U:Ljava/lang/Object;>(TU;)L1java/util/concurrent/CompletionStage<TU;>;

**Pull request merged**